aiida/cmdline/commands/rehash.py

```diff
@@ -7,63 +7,59 @@
 # For further information on the license, see the LICENSE.txt file        #
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
-import sys
-
 import click
 
-from aiida import try_load_dbenv
 from aiida.cmdline.baseclass import VerdiCommand
+from aiida.cmdline.commands import verdi_rehash
+from aiida.cmdline.params import arguments
+from aiida.cmdline.utils import decorators, echo
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
 
 
 class Rehash(VerdiCommand):
     """
-    Re-hash all nodes.
+    Re-hash nodes filtered by identifier and or node class
     """
 
     def run(self, *args):
-        ctx = _rehash_cmd.make_context('rehash', list(args))
+        ctx = rehash.make_context('rehash', list(args))
         with ctx:
-            _rehash_cmd.invoke(ctx)
+            rehash.invoke(ctx)
 
-    def complete(self, subargs_idx, subargs):
-        """
-        No completion after 'verdi rehash'.
-        """
-        print ""
 
-
-@click.command('rehash')
-@click.option('--all', '-a', is_flag=True, help='Rehash all nodes of the given Node class.')
-@click.option('--class-name', type=str, default='aiida.orm.node.Node',
-              help='Restrict nodes which are re-hashed to instances of this class.')
-@click.argument('pks', type=int, nargs=-1)
-def _rehash_cmd(all, class_name, pks):
-    try_load_dbenv()
+@verdi_rehash.command('rehash', context_settings=CONTEXT_SETTINGS)
+@arguments.NODES()
+@click.option('-e', '--entry-point', 'entry_point_string', type=click.STRING, default='aiida.node:node',
```

Comment:
Would it make sense to extend `aidia.cmdline.params.types.plugin.PluginParamType` to cover this more general case, i.e. to allow multiple categories and narrow down the completion results based on the start of the category in that case. 

I am not sure, but it looks like it might come in handy in other commands

Reply:
Yeah probably, the auto-complete and automatic validation would certainly be a big plus. In this case it does not even need to support multiple categories but we could generalize it to support that. Should we hardcode the `aiida` prefix for the category though? Won't this make it impossible to use entry points of other plugins?